x/btcstaking/keeper/finality_providers.go

```diff
@@ -68,6 +68,46 @@ func (k Keeper) SlashFinalityProvider(ctx context.Context, fpBTCPK []byte) error
 	return nil
 }
 
+// SlashConsumerFinalityProvider slashes a consumer finality provider with the given PK
+func (k Keeper) SlashConsumerFinalityProvider(ctx context.Context, consumerID string, fpBTCPK *bbn.BIP340PubKey) error {
```

Comment:
Do we need to emit an event so that the vigilante gets notified and slash?

Reply:
Also we might need to be careful for error handling here. Sometimes this might not be the error in Babylon, but consumer side reports false alarms. For example, it's possible that consumer side reports a slashed FP twice and L81 is triggered, making Babylon panick

Reply:
> hmm good point, i think for now its not needed as vigilante/btc-slasher is not subscribing to consumer tx events, can add a TODO for now but later would vigilante also monitor consumer chains?

How about we do this in this PR? This is a small change and vigilante in fact does not distinguish between Babylon/consumer tx events. So as long as we emit using code [here](https://github.com/babylonlabs-io/babylon/blob/1d97ebd1b94409083ae06da65e2289506c770e1f/x/finality/keeper/msg_server.go#L247-L251) vigilante shall work

Reply:
Ah I see, all good. Somehow I missed it, sorry :sweat_smile: